dialer.go

```diff
@@ -68,175 +63,172 @@ func NewDialer(opts *DialerOpts) Dialer {
 	if opts.WindowSize <= 0 {
 		opts.WindowSize = defaultWindowSize
 	}
-	if opts.MaxLiveConns <= 0 {
-		opts.MaxLiveConns = 1
+	if opts.LiveConns <= 0 {
+		opts.LiveConns = 2
```

Comment:
The higher the number of live conns, the less likely we are to coalesce frames. Any reason not to default to 1?

Reply:
The second connection has a much shorter dial timeout to handle temporary network disconnections or jumps in congestion to ensure lantern recovers more quickly.

The inclusion of that second connection also just gives it some redundancy such that it's more likely one of the connections will be functioning especially at congested times.

So we could switch to a single connection, but we'd need to either revert to the previous code to ensure liveness or come up with a new approach. To me, though, the advantages outweigh the disadvantages.

Reply:
> The inclusion of that second connection also just gives it some redundancy such that it's more likely one of the connections will be functioning especially at congested times.

If only one of the connections is actually functioning, what keeps us from picking the wrong connection for new streams?

Reply:
Sessions are only available when their TCP connections are active and the session is established. Streams certainly can and often do fail after that point, but the second connection at least gives dialers something to work with in the case the second connection has fully failed.